maven-core/src/test/java/org/apache/maven/classrealm/DefaultClassRealmManagerTest.java

```diff
@@ -0,0 +1,164 @@
+import org.eclipse.aether.artifact.Artifact;
+import org.junit.jupiter.api.Test;
+import org.mockito.InOrder;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.endsWith;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.calls;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Sebastien Doyon
+ */
+class DefaultClassRealmManagerTest {
+
+    private DefaultClassRealmManager newDefaultClassRealmManager(PlexusContainer container) {
+        HashSet<String> exportedPackages = new HashSet<String>();
+        exportedPackages.add("group1:artifact1");
+
+        return new DefaultClassRealmManager(
+                container,
+                new ArrayList<ClassRealmManagerDelegate>(),
+                new CoreExports(new ClassRealm(null, "test", null), new HashSet<String>(), exportedPackages));
+    }
+
+    private List<Artifact> newTestArtifactList() {
+        List<Artifact> artifacts = new ArrayList<Artifact>();
+
+        Artifact artifact = mock(Artifact.class);
+        when(artifact.getFile()).thenReturn(new File(new File("local/repository"), "some/path"));
+        when(artifact.getGroupId()).thenReturn("group1");
+        when(artifact.getArtifactId()).thenReturn("artifact1");
+        when(artifact.getExtension()).thenReturn("ext");
+        when(artifact.getClassifier()).thenReturn("classifier1");
+        when(artifact.getVersion()).thenReturn("1");
+        artifacts.add(artifact);
+
+        Artifact artifact2 = mock(Artifact.class);
+        when(artifact2.getFile()).thenReturn(null);
+        when(artifact2.getGroupId()).thenReturn("group1");
+        when(artifact2.getArtifactId()).thenReturn("artifact2");
+        when(artifact2.getExtension()).thenReturn("ext");
+        when(artifact2.getClassifier()).thenReturn("classifier1");
+        when(artifact2.getVersion()).thenReturn("1");
+        artifacts.add(artifact2);
+
+        return artifacts;
+    }
+
+    private Model newTestModel() {
+        Model model = new Model();
+        model.setGroupId("modelGroup1");
+        model.setArtifactId("modelArtifact1");
+        model.setVersion("modelVersion1");
+
+        return model;
+    }
+
+    @Test
+    void testDebugEnabled() throws PlexusContainerException {
+        Logger logger = mock(Logger.class);
+        when(logger.isDebugEnabled()).thenReturn(true);
+
+        DefaultClassRealmManager classRealmManager;
+        ClassRealm classRealm;
+
+        InOrder verifier = inOrder(logger);
+
+        PlexusContainer container = new DefaultPlexusContainer();
+
+        try (MockedStatic<LoggerFactory> mockedLoggerFactory = Mockito.mockStatic(LoggerFactory.class)) {
```

Comment:
In general, I'm skeptical of testing log messages. I tend to think of them as side effects that can change pretty freely, and not part of a public API contract that should be tested.

Reply:
I agree with the principle. I did this test class because in the it was specified in the contribution guidelines : "Make sure you have added the necessary tests (JUnit/[Core IT tests](https://maven.apache.org/core-its/core-it-suite/)) for your changes.". And I found at least one other test example that was testing log messages, the DefaultToolchainTest class. I don't know if any external tools may be expecting maven to log in a specific format, like Maven Daemon? If you think the tests are a bad idea, I am willing to remove, just tell me.